llpc/util/llpcElfWriter.cpp

```diff
@@ -1209,6 +1209,42 @@ bool getDeviceIndexRelocationValue(Context *context, RelocationEntry relocEntry,
   return true;
 }
 
+// =====================================================================================================================
+// Get value of a numSamples relocation ($numSamples symbol).
+//
+// @param context : [in] Pipeline compilation context
+// @param relocEntry : The relocation entry to fixup
+// @param isGraphicsPipeline : Whether we are processing a compute or graphics pipeline
+// @param value : [out] The value of the relocation
+bool getNumSamplesRelocationValue(Context *context, RelocationEntry relocEntry, bool isGraphicsPipeline,
+                                  unsigned *value) {
+  auto pipelineBuildInfo = context->getPipelineContext()->getPipelineBuildInfo();
+  if (isGraphicsPipeline) {
+    *value = static_cast<const GraphicsPipelineBuildInfo *>(pipelineBuildInfo)->rsState.numSamples;
+  } else {
+    llvm_unreachable("num samples relocation is for graphics pipeline only.");
+  }
```

Comment:
Minor note: For stuff like this, you could just say `assert(isGraphicsPipeline)` or, if you want to be more verbose about it, `assert(isGraphicsPipeline && "num samples relocation is for graphics pipeline only");`. This is a common pattern in LLVM code.